### PR TITLE
fix_get_transaction_receipt

### PR DESCRIPTION
**Pull request by regcostajr**

### CODE REVIEW

File: lib/Net/Async/Blockchain/ETH.pm [hunk 1]
```diff
@@ -276,12 +284,12 @@ async sub _set_transaction_type {
     my $accounts_response = await $self->rpc_client->accounts();
     return undef unless $accounts_response;
 
-    my %accounts = map { $_ => 1 } $accounts_response->@*;
+    my %accounts = map { lc($_) => 1 } $accounts_response->@*;
```
Comment (turnstep) [inline lib/Net/Async/Blockchain/ETH.pm]: All this lowercasing seems fragile and easy to forget somewhere. Is it possible to do this somewhere higher up, or is the case significant in other uses?

File: lib/Net/Async/Blockchain/ETH.pm [hunk 2]
```diff
@@ -325,11 +333,22 @@ async sub _check_contract_transaction {
...
 
+    # even if 1 block confirmation sometimes the node still take some seconds to update
+    # the info and the eth_getTransactionReceipt command returns a empty result, to handle
+    # this we check at least 5 times for the transaction receipt, it's enough time to
+    # receive the response.
+    # if the logs are still empty after all we going to handle this a non contract transaction.
     my $receipt = await $self->rpc_client->get_transaction_receipt($transaction->hash);
-    my $logs    = $receipt->{logs};
+    my $retries = 0;
+    while (!$receipt->{logs} && $retries < 5) {
```
Comment (turnstep) [inline lib/Net/Async/Blockchain/ETH.pm]: Be nice to have 5 as a constant at the top of the file

File: lib/Net/Async/Blockchain/ETH.pm (hunk 1 above)
Comment (regcostajr) [reply lib/Net/Async/Blockchain/ETH.pm]: @turnstep this is a double-check the node will return the transaction in lowercase already, I'm adding this just for precaution.

File: lib/Net/Async/Blockchain/ETH.pm [hunk 3]
```diff
@@ -269,32 +276,27 @@ hashref from an array of L<Net::Async::Blockchain::Transaction>
...
-            $transaction->{type} = 'internal';
-        } elsif ($from) {
-            $transaction->{type} = 'sent';
-        } elsif ($to) {
-            $transaction->{type} = 'receive';
-        } else {
-            next;
-        }
-        push(@node_transactions, $transaction) if $transaction->type;
+    my %accounts = map { lc($_) => 1 } $accounts_response->@*;
```
Comment (zakame) [inline lib/Net/Async/Blockchain/ETH.pm]: Using postderefs need Perl 5.24, so `cpanfile` here needs an update.

**Pull request merged**